gadgets/trace_tcpconnect/program.bpf.c

```diff
@@ -64,15 +69,26 @@ const volatile __u64 targ_min_latency_ns = 0;
 #define AF_INET6 10
 
 // sockets_per_process keeps track of the sockets between:
-// - kprobe enter_tcp_connect
-// - kretprobe exit_tcp_connect
+// - kprobe inet_stream_connect
+// - kretprobe inet_stream_connect
 struct {
 	__uint(type, BPF_MAP_TYPE_HASH);
 	__uint(max_entries, MAX_ENTRIES);
 	__type(key, u32); // tid
 	__type(value, struct sock *);
 } sockets_per_process SEC(".maps");
 
+// src_dst_per_process keeps track of the src and dst information
+// between tcp_v4/6_connect and the return of inet_stream_connect
+// Since tcp_v4/6_connect is directly inside of inet_stream_connect,
+// we do not need to keep a huge number of entries.
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 128);
```

Comment:
I don't understand why max_entries is different between the two maps `sockets_per_process` and `src_dst_per_process`.

Reply:
It might be a (premature) optimization.

`sockets_per_process` stores a pointer, whereas `src_dst_per_process` is at least 42 bytes large. `src_dst_per_process` gets written into after the `SYN` package gets sent, but with synchronous sockets the waiting is still done afterwards.
So we might need to have the larger buffer?